src/main/java/software/amazon/awssdk/eventstreamrpc/OperationContinuationHandler.java

```diff
@@ -364,28 +396,31 @@ private void handleAndSendError(Throwable throwable) {
             throwable = throwable.getCause();
         }
         if (throwable instanceof EventStreamOperationError) {
-            //We do not check if the specific exception thrown is a part of the core service?
+            // We do not check if the specific exception thrown is a part of the core
+            // service?
             sendModeledError((EventStreamOperationError) throwable);
             invokeAfterHandleRequest();
         } else {
             final List<Header> responseHeaders = new ArrayList<>(1);
             byte[] outputPayload = "InternalServerError".getBytes(StandardCharsets.UTF_8);
             responseHeaders.add(Header.createHeader(EventStreamRPCServiceModel.CONTENT_TYPE_HEADER,
                     EventStreamRPCServiceModel.CONTENT_TYPE_APPLICATION_TEXT));
-            //are there any exceptions we wouldn't want to return a generic server fault?
-            //this is the kind of exception that should be logged with a request ID especially in a server-client context
+            // are there any exceptions we wouldn't want to return a generic server fault?
+            // this is the kind of exception that should be logged with a request ID
+            // especially in a server-client context
             LOGGER.error("[{}] operation threw unexpected {}: {}", getOperationName(),
                     throwable.getClass().getCanonicalName(), throwable.getMessage());
 
             continuation.sendMessage(responseHeaders, outputPayload, MessageType.ApplicationError,
                     MessageFlags.TerminateStream.getByteValue()).whenComplete((res, ex) -> {
-                if (ex != null) {
-                    LOGGER.error(ex.getClass().getName() + " sending error response message: " + ex.getMessage());
-                } else {
-                    LOGGER.trace("Error response successfully sent");
-                }
-                continuation.close();
-            });
+                        if (ex != null) {
+                            LOGGER.error(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using String.format(), string concatenation, or StringBuilder in logging statements can lead to unnecessary string formatting, even when the log level is disabled, potentially impacting application performance. SLF4J offers parameterized logging, which defers string formatting until the log message is actually needed. This approach can significantly improve efficiency, especially in high-throughput scenarios or when certain log levels are frequently disabled. Learn more - https://cwe.mitre.org/data/definitions/1176.html